src/status_im/ui/screens/profile/events.cljs

```diff
@@ -1,58 +1,80 @@
+         (img->base64 path on-success on-error))))))
+
+(handlers/register-handler-fx
+  :profile/send-transaction
+  (fn [_ [_ chat-id]]
+    {:dispatch-n [[:clear-seq-arguments]
+                  [:navigate-to :chat chat-id]
+                  ;; TODO https://github.com/status-im/status-react/issues/1621
+                  [:select-chat-input-command {:name "send"}]]}))
+
+(handlers/register-handler-fx
+  :profile/send-message
+  (fn [_ [_ identity]]
+    (when identity
+      {:dispatch [:navigation-replace :chat identity]})))
+
+(handlers/register-handler-fx
+  :my-profile/edit
+  (fn [{:keys [db]} [_ edit-type edit-value]]
+    (let [current-account-id (:current-account-id db)
+          current-account (select-keys (get-in db [:accounts current-account-id])
+                                       [:name :photo-path :status])
+          new-db (-> db
+                     (update-in [:my-profile/edit] merge current-account)
+                     (assoc-in [:my-profile/edit :edit-status?] (= edit-type :status true)))]
+      {:db new-db
+       :dispatch [:navigate-to :edit-my-profile]})))
+
+(handlers/register-handler-fx
+  :my-profile/update-status
+  (fn [{:keys [db]} [_ new-status]]
+    (when-not (string/blank? new-status)
+      {:db (assoc-in db [:my-profile/edit :status] new-status)
+       :dispatch-n [[:check-status-change new-status]
+                    [:account-update {:status new-status}]]})))
+
+(handlers/register-handler-fx
+  :my-profile/update-phone-number
   ;; Switch user to the console issuing the !phone command automatically to let him change his phone number.
   ;; We allow to change phone number only from console because this requires entering SMS verification code.
-  (handlers/side-effect!
-   (fn [db _]
-     (re-frame/dispatch [:navigate-to :chat console-chat-id])
-     (js/setTimeout #(re-frame/dispatch [:select-chat-input-command {:name "phone"}]) 500))))
-
-(handlers/register-handler
-  :open-chat-with-the-send-transaction
-  (handlers/side-effect!
-   (fn [db [_ chat-id]]
-     (re-frame/dispatch [:clear-seq-arguments])
-     (re-frame/dispatch [:navigate-to :chat chat-id])
-     (js/setTimeout #(re-frame/dispatch [:select-chat-input-command {:name "send"}]) 500))))
-
-(defn prepare-edit-profile
-  [{:keys [current-account-id] :as db} _]
-  (let [current-account (select-keys (get-in db [:accounts current-account-id])
-                                     [:name :photo-path :status])]
-    (update-in db [:profile-edit] merge current-account)))
-
-(defn open-edit-profile [_ _]
-  (re-frame/dispatch [:navigate-to :edit-my-profile]))
-
-(handlers/register-handler
-  :open-edit-my-profile
-  (handlers/handlers->
-   prepare-edit-profile
-   open-edit-profile))
+  (fn [_ _]
+    {:dispatch-n [[:navigate-to :chat console-chat-id]
+                  [:select-chat-input-command {:name "phone"}]]}))
+
+(handlers/register-handler-fx
+  :my-profile/update-picture
+  (fn [{:keys [db]} [this-event base64]]
+    (if base64
```

Comment:
I agree with @jeluard on splitting this into two distinct handlers. While it's neat that the `:my-profile/update` picture can cause effect which in turn dispatches the same event again with different args (to be honest, for me, this is the first time where having dispatch keyword in the event vector argument is of clear utility :) ), I think that having them separate is still easier to reason and easier to extend in the future.
But not a big thing, rename `base64` to `base64-image` or `image-data` would be sufficient for me.